R/compute_metrics.R

```diff
@@ -238,11 +240,10 @@ computeSCR <- function(object,
         } else {
             sampIdx <- grep(samplePattern, annot)
         }
-        if (!length(sampIdx)) stop("No match found with 'samplePattern = ", 
-                                   samplePattern, "'.")
         carrIdx <- grep(carrierPattern, annot)
-        if (!length(carrIdx)) stop("No match found with 'carrierPattern = ", 
-                                   carrierPattern, "'.")
+        if (!length(sampIdx)) sampleNotFound <- c(sampleNotFound, ii)
+        if (!length(carrIdx)) carrierNotFound <- c(carrierNotFound, ii)
+        if (length(c(sampIdx, carrIdx)) != 2) next
```

Comment:
I think this line is wrong. If you run the code in the `scp.Rmd` vignette, you'll see that `next` is called for the 3 assays, and no ratio is ever computer and added to the `rowData`.